lib/cosmos/system/target.rb

```diff
@@ -123,12 +122,28 @@ def process_file(filename)
         when 'REQUIRE'
           usage = "#{keyword} <FILENAME>"
           parser.verify_num_parameters(1, 1, usage)
+          found = false
           begin
-            Cosmos.require_file(parameters[0])
+            # Determine if this file is in our target lib and should be namespaced
+            filename = File.join(@dir, 'lib', parameters[0])
+            if File.exist? filename
+              const = find_or_create_const(@name)
+              const.module_eval(File.read(File.join(@dir, 'lib', parameters[0])))
+              found = true
+              @requires << filename
```

Comment:
I don't think this line is needed.   Seems like you are double booking into @requires.